tests/utils.go

```diff
@@ -2080,31 +2092,17 @@ func NewRandomFedoraVMIWithDmidecode() *v1.VirtualMachineInstance {
 }
 
 func GetGuestAgentUserData() string {
-	virtClient, err := kubecli.GetKubevirtClient()
-	PanicOnError(err)
-
-	dnsServerIP, err := getClusterDnsServiceIP(virtClient)
-	PanicOnError(err)
-
-	ipv6UserDataString := ""
-	if netutils.IsIPv6String(dnsServerIP) {
-		ipv6UserDataString = fmt.Sprintf(`sudo ip -6 addr add fd10:0:2::2/120 dev eth0
-                             for i in {1..20}; do sudo ip -6 route add default via fd10:0:2::1 src fd10:0:2::2 && break || sleep 0.1; done
-                             echo "nameserver %s" >> /etc/resolv.conf`, dnsServerIP)
-	}
 	return fmt.Sprintf(`#!/bin/bash
                 echo "fedora" |passwd fedora --stdin
-                systemctl disable NetworkManager
-                systemctl stop NetworkManager
-                %s
                 mkdir -p /usr/local/bin
+                sleep 1
```

Comment:
NetworkManager is being started before cloud-init, and abducts the cloud init interface when it created.

The NetworkManager-wait-online.service service ends up waiting 30 seconds for NetworkManager to finish configuring the interface. Only once this fails can cloud-init proceed to userdata - hence, waiting is necessary.

This is caused by canonical/cloud-init@8888ca1, which was introduced in an attempt to fix https://bugs.launchpad.net/cloud-init/+bug/1843334.

Reply:
It is not clear how the `sleep` added here relates to the issue presented here.
The issue relates to a long VM boot time, but post this wait, the cloud-init network-data kicks in, and NM should apply the configuration. This sleep here is post that configuration and may result from the NM not really finishing it.

I guess we should use a retry on the first `curl` fetch.